R/shared-methods.R

```diff
@@ -420,6 +427,17 @@ CountryDataClass <- R6::R6Class("CountryDataClass",
         }
       }
       super$filter()
+    },
+
+    #' @description Process function for national data sources. Runs
+    #' `DataClass` process first before filling `country_info`.
+    process = function() {
+      super$process()
+      if (!(is.na(self$supported_region_names[["1"]]))) {
```

Comment:
I think we want this in clean common because we want to be able to inform filtering prior to processing?

Reply:
isn't it okay to have it fall all national data sources?

Reply:
So use case would be. 

```
uk <- Google$new()
uk$download()
uk$clean()
uk$show_countries()
uk$filter("United Kingdom")
uk$process()
uk$return()
```

Where show_countries is simply printing the stored country_info filed? 


```

Reply:
or could just call the uk$country_field directly?

Reply:
tbh it could happen in download as you have the index.csv that has all of this in?

Reply:
Agreed, I will move this to download.

Reply:
Sorry, i'm getting knotted up. Do we want this just for Google? or for all national datasets? If just for google then `$show_countries()` and the field `country_info` would be specific to Google, and we could have it set from download using index.csv. But we could apply this to all national datasets if we moved it before processing (and put the methods / fields in `CountryDataClass`. If we put in the cleaning step it would not take into account filtering.

Reply:
I assumed just for google and JHU...... Aren't we call with it not updating with filtering if we just want it to show the available info? I mean JHU and Google could inherit from a small class of their own I guess but that is getting quite turtle all the way down.